.github/workflows/build-and-deploy.yml

```
@@ -0,0 +1,43 @@
+name: "Build and Deploy"
+on:
+  push: {}
+  pull_request: {}
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - uses: cachix/install-nix-action@v23
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+    - uses: cachix/cachix-action@v12
+      with:
+        name: nix-dev
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+
+    - name: Build
+      run: |
+        nix-build
+
+        # The netlify action doesn't follow symlinks properly.
+        mkdir ./dist
+        cp -RL ./result/* ./dist/
+
+    - name: Deploy to Netlify
+      uses: nwtgck/actions-netlify@v2.0
```

```suggestion
      uses: nwtgck/actions-netlify@v2.1.0
```

Note that it has a new option `enable-github-deployment`, which defaults to `true`. I think it's fine, but FYI

---

Also I noticed there's an [official action](https://github.com/netlify/actions/tree/master/build), but it's been a while since it was updated. Is the one above better?

---

The official action just tells Netlify to perform a new build via their own build service (similar to Cloudflare). That wouldn't work here since you want to be able to build via Nix.

Will update to 2.1.0.